### PR TITLE
Update find endpoint to allow filtering by request args

### PR DESCRIPTION
**Pull request by jplulu**
Also update spinc cli find to work with the new filter.

- [x] Test in staging

### CODE REVIEW

File: request-manager/request/manager.go [hunk 1]
```diff
@@ -540,39 +540,46 @@ func (m *manager) GetWithJC(requestId string) (proto.Request, error) {
...
 		for _, state := range filter.States {
 			values = append(values, state)
 		}
 	}
+	if len(filter.Args) != 0 {
+		for arg, val := range filter.Args {
+			fields = append(fields, "ra.create_request LIKE CONCAT('%\"', ?, '\":\"', ?, '\"%')")
+			values = append(values, arg, val)
+		}
+	}
```
Comment (felixp-square) [inline request-manager/request/manager.go:568]: Could you add a test for the args filter to [TestFind](https://github.com/square/spincycle/blob/cf48fe75816ae2e0932844dd8069e1757329d2d9/request-manager/request/manager_test.go#L662)?

File: spinc/cmd/find.go [hunk 2]
```diff
@@ -255,6 +268,7 @@ Filters:
   type        type of request to return
   states      comma-separated list of request states to include
   user        return only requests made by this user
+  requestArgs return requests made with specific args (format: arg1=value1,arg2=value2)
```
Comment (felixp-square) [inline spinc/cmd/find.go:271]: `requestArgs` -> `args` I think, cause you're grabbing "args" from the map at line 108

**Review by felixp-square — COMMENTED** [review]
couple of minor comments but overall lgtm

File: request-manager/request/manager.go (hunk 1 above)
Comment (jplulu) [reply request-manager/request/manager.go:568]: I was looking to add a test for this but it seems that the existing test framework does not use/populate the `request_archives` table. So args will always be empty.

File: spinc/cmd/find.go (hunk 2 above)
Comment (jplulu) [reply spinc/cmd/find.go]: done

File: request-manager/request/manager.go (hunk 1 above)
Comment (jplulu) [reply request-manager/request/manager.go:568]: I'm not sure how to update it to populate `request_archives` table

**Review by felixp-square — APPROVED** [review]

**Pull request merged**